src/tests/ftest/aggregation/aggregation_throttling.py

```diff
@@ -0,0 +1,151 @@
+        # Enable the aggregation
+        self.pool.set_property("reclaim", "time")
+
+        # Run ior the third time while the aggregation of first two runs
+        # are running in the background.
+        out = self.run_ior_with_pool(update=False)
+        metric_after_aggregate = IorCommand.get_ior_metrics(out)
+
+        # When DAOS-5057 is fixed, adjust the percentage. For now,
+        # keep it at 30 %
+        expected_perf_diff = 30.0
+
+        self.verify_performance(metric_before_aggregate,
+                                metric_after_aggregate,
+                                0, # write_perf
+                                expected_perf_diff) # 30% perf difference
+
+        self.verify_performance(metric_before_aggregate,
+                                metric_after_aggregate,
+                                1, # read_perf
+                                expected_perf_diff)
+
+    def verify_performance(self, before_metric, after_metric, read_write_idx,
+                           expected_perf_diff):
+        """Verify the after_metric read/write performances are within
+           +/- expected_perf_diff % of before_metric_performance.
+
+        Args:
+           before_metric (IorMetrics): ior metrics before aggregation
+           after_metric (IorMetrics): ior metrics in concurrent with aggregation
+           read_write_idx (int): Read (1) or Write (0) index in the IorMetrics
+           expected_perf_diff (float): Expected performance difference between
+                                       before_metric and after_metric.
+        """
+
+        self.log.info("\n\n%s Performance" %
+                      "Read" if read_write_idx else "Write")
+
+        max_mib = int(IorMetrics.Max_MiB)
+        min_mib = int(IorMetrics.Min_MiB)
+        mean_mib = int(IorMetrics.Mean_MiB)
+
+        max_prev = float(before_metric[read_write_idx][max_mib])
+        max_curr = float(after_metric[read_write_idx][max_mib])
+        self.log.info("max_prev = %f, max_curr = %f" % (
+            max_prev, max_curr))
+
+        min_prev = float(before_metric[read_write_idx][min_mib])
+        min_curr = float(after_metric[read_write_idx][min_mib])
+        self.log.info("min_prev = %f, min_curr = %f" % (
+            min_prev, min_curr))
+
+        mean_prev = float(before_metric[read_write_idx][mean_mib])
+        mean_curr = float(after_metric[read_write_idx][mean_mib])
+        self.log.info("mean_prev = %f, mean_curr = %f" % (
+            mean_prev, mean_curr))
+
+        max_perf_diff = (abs(max_prev - max_curr)/max_prev) * 100
+        min_perf_diff = (abs(min_prev - min_curr)/min_prev) * 100
+        mean_perf_diff = (abs(mean_prev - mean_curr)/mean_prev) * 100
+
+        self.log.info("Max perf diff %f < %f" % (max_perf_diff,
+                                                 expected_perf_diff))
+        self.assertTrue(max_perf_diff < expected_perf_diff)
+        self.log.info("Min perf diff %f < %f" % (min_perf_diff,
```

Comment:
(pylint-logging-not-lazy) Specify string format arguments as logging function parameters